### DISCUSSION TITLE
A way to get all messages in thread by id?

### DISCUSSION THREAD

**Discussion by lokize** [discordjs/discord.js#9370] [Q&A] [answered]
Hi, is there any way to get all messages whit user and avatar in thread by id?

i using

const channel = await client.channels.fetch("id");

and returned all content in this channel thread but only topics not messages...

**Comment by BenjammingKirby** [comment, marked as answer]
not really through a single fetch (all the time)
your `channel.messages.fetch({limit:100})` is going to fetch the last 100 messages (as the max limit is 100), so to fetch more, you need to fetch more and do some pagination (using the before and after parameter)
if there is a huge number of messages, it can turn into api spam real quick

**Reply by lokize** [reply]
channel.messages.fetch not worked "TypeError: Cannot read properties of undefined (reading 'fetch')"

**Reply by BenjammingKirby** [reply]
> channel.messages.fetch not worked "TypeError: Cannot read properties of undefined (reading 'fetch')"

then simply, the `channel` you have isn't a threadchannel, make sure it is
log it

**Reply by lokize** [reply]
thanks, yes is a forum channel... how to get messages from forum channel?

**Reply by Jiralite** [reply]
A forum channel does not contain messages. A forum channel contains threads. You will have to fetch (in one way or another) those threads and then fetch messages on them.

https://old.discordjs.dev/#/docs/discord.js/14.9.0/class/ForumChannel?scrollTo=threads

**Reply by lokize** [reply]
thanks guys for helping! worked!
